etcd/requests.go

```diff
@@ -303,8 +303,8 @@ func DefaultCheckRetry(cluster *Cluster, numReqs int, lastResp http.Response,
 	err error) error {
 
 	if numReqs > 2*len(cluster.Machines) {
-		return newError(ErrCodeEtcdNotReachable,
-			"Tried to connect to each peer twice and failed", 0)
+		errStr := fmt.Sprintf("Tried to connect to members %v twice and failed: %v", cluster.Machines, err)
```

Comment:
It's still weird that you would even try twice if the error is not a "try again" error or a connection failure.  That the string says "tried to connect ... and failed" implies that there is an error with the connection, but there isn't.  It's better than before, but only in the most minimal possible way.